src/jbuild_types.ml

```diff
@@ -678,6 +678,60 @@ module Rule = struct
   let ocamlyacc_vjs = ocamlyacc_v1
 end
 
+module Menhir = struct
+  type t =
+    { base : string option
+    ; flags : string list
+    ; modules: string list
+    }
+
+  let v1 =
+    record
+      (field_o "base" string >>= fun base ->
+       field "flags" (list string) ~default:[] >>= fun flags ->
```

Comment:
hmm, no indeed. I might be missing something, but it seems to me that here `flags` is a `string list` and at the beginning of `v1_to_rule` we convert it to a `String_with_vars.t list`. We should be able to parse it immediately as a `String_with_vars.t list`?